storm/src/main/storm/mesos/resources/OfferResources.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package storm.mesos.resources;
+
+import backtype.storm.scheduler.TopologyDetails;
+import org.apache.mesos.Protos;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import storm.mesos.util.MesosCommon;
+import storm.mesos.util.PrettyProtobuf;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// TODO(ksoundararaj): a better name? - OfferAggregator?
+public class OfferResources {
+
+  private final Logger log = LoggerFactory.getLogger(OfferResources.class);
+  private Map<ResourceType, Resource> availableResources;
+
+  private List<Protos.Offer> offerList = new ArrayList<Protos.Offer>();
+
+  private final String hostName;
+
+  private Protos.SlaveID slaveID;
+
+  private void initializeAvailableResources() {
+    availableResources = new HashMap<>();
+    availableResources.put(ResourceType.CPU, new ScalarResource(ResourceType.CPU));
+    availableResources.put(ResourceType.MEM, new ScalarResource(ResourceType.MEM));
+    availableResources.put(ResourceType.PORTS, new RangeResource(ResourceType.PORTS));
+  }
+
+  public OfferResources(String hostName) {
+    this.hostName = hostName;
+    initializeAvailableResources();
+  }
+
+  public OfferResources(Protos.Offer offer) {
+    initializeAvailableResources();
+    if (slaveID == null) {
```

Comment:
This check seems unnecessary since this is a constructor and slaveID can't be anything but null at this point.